app/upload/video/camera.tsx

```diff
@@ -0,0 +1,191 @@
+import BottomRow from '../../../components/upload/video/BottomRow';
+import Camera from '../../../components/upload/video/Camera';
+
+export default function Page() {
+    const cameraRef = useRef<CameraView>(null);
+    const [cameraPermission, requestCameraPermission] = useCameraPermissions();
+    const [mediaLibraryPermission, requestMediaLibraryPermission] = useMediaLibraryPermissions();
+    const [useMicrophonePermission, requestMicrophonePermission] = useMicrophonePermissions();
+    const [isRecording, setIsRecording] = useState(false);
+    const [videoUri, setVideoUri] = useState<string | null>(null);
+    const [facing, setFacing] = useState<CameraType>('back');
+    const [flash, setFlash] = useState<FlashMode>('off');
+    const [zoom, setZoom] = useState<number>(0);
+    const [lastZoom, setLastZoom] = useState<number>(0);
+    const [timer, setTimer] = useState<number>(60)
+    const timerRef = useRef<NodeJS.Timeout | null>(null);
+    const currentDate = new Date();
+    const currentDateIso = `${currentDate.getFullYear()}-${currentDate.getMonth() + 1}-${currentDate.getDate()}T00:00:00`;
+
+    useEffect(() => {
+        if (isRecording) {
+            timerRef.current = setInterval(() => {
+                setTimer(prev => {
+                    if (prev <= 1) {
+                        clearInterval(timerRef.current!);
+                        setIsRecording(false);
+                        return 60;
+                    }
+                    return prev - 1;
+                });
+            }, 1000);
+        } else {
+            if (timerRef.current) {
+                clearInterval(timerRef.current);
+                setTimer(60);
+            }
+        }
+
+        return () => {
+            if (timerRef.current) {
+                clearInterval(timerRef.current);
+            }
+        };
+    }, [isRecording]);
+
+
+    if (!cameraPermission || !mediaLibraryPermission || !useMicrophonePermission) {
+        //Change to loading state
+        return <View />
+    }
+
+    async function requestPermission() {
+        requestCameraPermission();
+        requestMediaLibraryPermission();
+        requestMicrophonePermission();
+    }
+
+    if (!cameraPermission.granted || !mediaLibraryPermission.granted || !useMicrophonePermission.granted) {
+        return (
+            <View className =  "w-full h-full flex justify-center items-center gap-y-10">
+                <Text>We need your permission to use the Camera!</Text>
+                <Pressable onPress={requestPermission} className = "bg-black px-3 py-2 rounded-2xl">
+                    <Text className = " text-white font-medium">Request permission</Text>
+                </Pressable>
+            </View>
+        );
+    }
+
+    function toggleFacing() {
+        setFacing(facing === 'back' ? 'front' : 'back');
+    }
+
+    function toggleFlash() {
+        setFlash(flash === 'off' ? 'on' : 'off');
+    }
+
+    async function toggleRecording() {
```

Comment:
I think it would be worthwhile to maybe make these functions with a try catch block as I'd imagine that we might have some sort of connection errors with these async functions.